be/src/runtime/query_context.cpp

```diff
@@ -55,6 +63,14 @@ QueryContext::~QueryContext() {
     }
 
     LOG_INFO("Query {} deconstructed, {}", print_id(_query_id), mem_tracker_msg);
+    // Not release the the thread token in query context's dector method, because the query
+    // conext may be dectored in the thread token it self. It is very dangerous and may core.
+    // And also thread token need shutdown, it may take some time, may cause the thread that
+    // release the token hang, the thread maybe a pipeline task scheduler thread.
+    if (_thread_token) {
+        static_cast<void>(ExecEnv::GetInstance()->lazy_release_obj_pool()->submit(
```

Comment:
When will _thread_token be released?

Reply:
it is moved to release token task, when the task run finished, it will be released. 
currently, i did not find a better way. I will remove thread token logic in the future.